src/client/datascience/jupyter/kernels/helpers.ts

```diff
@@ -95,9 +95,10 @@ export function getKernelId(spec: IJupyterKernelSpec, interpreter?: PythonEnviro
         // Lets not assume that non-python kernels cannot have such issues
         argsForGenerationOfId = spec.argv.join('#').toLowerCase();
     }
-    return `${spec.id || ''}.${specName}.${getNormalizedInterpreterPath(spec.interpreterPath || spec.path)}.${
-        getNormalizedInterpreterPath(interpreter?.path) || ''
-    }.${argsForGenerationOfId}`;
+    const prefixForRemoteKernels = remoteBaseUrl ? `${remoteBaseUrl}.` : '';
+    return `${prefixForRemoteKernels}${spec.id || ''}.${specName}.${getNormalizedInterpreterPath(
```

Comment:
Give a unique id for remote kernels (include the base url)
Else if you start jupyter locally, they could have the same ids (as they point to the same kernelspec)

An added benefit is the fact that this now allows us to support multiple remotes at a time.

Reply:
This might be rather long? Seems a bit much? Like from azure it can be ```azure.westcoastus.en.mysillyloginname/jupyter/server/332```. I guess we can wait for people complaining though.

Reply:
Does long matter? I don't think that the kernel ID is user visible as all unless I'm off on this.

Reply:
Yeah it doesn't matter. I was thinking this was in the display. For the id it shouldn't matter.